### PR TITLE
Adding Bear 65 Ergo Keyboard To QMK

### PR DESCRIPTION
**Pull request by MudkipMao**
Jacky, the maker of the S7 has a new board and I ported it into QMK

## Description

Not much to say, please take a look at the changes. I have the board in hand and it's running nicely

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by MudkipMao** [comment]
Hi QMK maintainers, one thing I do want to ask is how we want to structure this folder. Jacky is the designer of the S7_Elephant which currently has its own folder. Should I consolidate both of these boards under 1 folder?

File: keyboards/bear_65/info.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "keyboard_name": "Bear 65",
+    "url": "https://qmk.fm/keyboards",
+    "maintainer": "qmk",
+    "width": 19.75,
+    "height": 6.014620257184662,
+    "layouts": {
+        "LAYOUT_all": {
+            "key_count": 69,
```
Comment (MudkipMao) [inline keyboards/bear_65/info.json:9]: I'll clean up the weird values in the JSON tonight

Comment (drashna) [inline keyboards/bear_65/info.json:9]:
```suggestion
```

File: keyboards/bear_65/info.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        },
+        "LAYOUT_full_bs": {
+            "key_count": 68,
```
Comment (drashna) [inline keyboards/bear_65/info.json:13]:
```suggestion
```

File: keyboards/bear_65/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN };
```
Comment (drashna) [inline keyboards/bear_65/keymaps/default/keymap.c:21]:
```suggestion
    _FN,
};
```

File: keyboards/bear_65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+
+KEY_LOCK_ENABLE = no 	    # Enable KC_LOCK support
+LTO_ENABLE = yes            # Use LTO flags to reduce firmware size
```
Comment (drashna) [inline keyboards/bear_65/rules.mk:28]:
```suggestion
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output

LTO_ENABLE = yes            # Use LTO flags to reduce firmware size
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bear_65/config.h [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xA13B // Jacky's vendor ID, arbitrarily selected by MudkipMao
+#define PRODUCT_ID      0x000A
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Jacky
+#define PRODUCT         Jacky Bear 65 Ergo
+#define DESCRIPTION     A gasket-mounted 65% Alice style board
```
Comment (fauxpark) [inline keyboards/bear_65/config.h:28]:
```suggestion
```

File: keyboards/bear_65/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+           KC_PGDN,   KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G,                KC_H, KC_J, KC_K, KC_L,  KC_SCLN,   KC_QUOT,   KC_ENT,                \
+                      KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B,                KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP,           \
+                      KC_LCTL,     KC_LALT, KC_SPC, KC_LGUI,                KC_SPC,     MO(_FN),                        KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [_FN] = LAYOUT_all(
+             RESET,   _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,          KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______,     _______, \
+            RGB_MOD,  _______, KC_BRID, KC_BRIU, _______, _______, _______,                 _______, _______, _______, _______, _______, RGB_VAD, RGB_VAI, _______,              \
+           RGB_TOG,   _______, KC_VOLD, KC_VOLU, KC_MUTE, _______, _______,                _______, _______, _______, _______,  _______,   _______,   _______,                \
+                      _______, KC_MPRV, KC_MPLY, KC_MNXT, _______, _______,                _______, _______, _______, _______, _______, _______, _______, _______,           \
```
Comment (fauxpark) [inline keyboards/bear_65/keymaps/default/keymap.c:36]:
```suggestion
             KC_ESC,  KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6,           KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC, KC_GRV,     KC_DEL,
            KC_PGUP,  KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T,                 KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS,
           KC_PGDN,   KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G,                KC_H, KC_J, KC_K, KC_L,  KC_SCLN,   KC_QUOT,   KC_ENT,
                      KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B,                KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP,
                      KC_LCTL,     KC_LALT, KC_SPC, KC_LGUI,                KC_SPC,     MO(_FN),                        KC_LEFT, KC_DOWN, KC_RGHT
    ),
    [_FN] = LAYOUT_all(
             RESET,   _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,          KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______,     _______,
            RGB_MOD,  _______, KC_BRID, KC_BRIU, _______, _______, _______,                 _______, _______, _______, _______, _______, RGB_VAD, RGB_VAI, _______,
           RGB_TOG,   _______, KC_VOLD, KC_VOLU, KC_MUTE, _______, _______,                _______, _______, _______, _______,  _______,   _______,   _______,
                      _______, KC_MPRV, KC_MPLY, KC_MNXT, _______, _______,                _______, _______, _______, _______, _______, _______, _______, _______,
```

File: keyboards/bear_65/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                      _______, _______, _______, _______, _______, _______,                _______, _______, _______, _______, _______, _______, _______, _______,           \
+                      _______,     _______, _______, _______,                _______,     _______,                        _______, _______,_______ 
+    ),
+    [3] = LAYOUT_full_bs(
+             _______,   _______, _______,   _______,   _______,   _______,   _______,   _______,          _______,   _______,   _______,   _______,  _______,  _______,  _______,     _______,      \
+            _______,  _______, _______, _______, _______, _______, _______,                 _______, _______, _______, _______, _______, _______, _______, _______,          \
+           _______,   _______, _______, _______, _______, _______, _______,                _______, _______, _______, _______,  _______,   _______,   _______,               \
+                      _______, _______, _______, _______, _______, _______,                _______, _______, _______, _______, _______, _______, _______, _______,           \
```
Comment (fauxpark) [inline keyboards/bear_65/keymaps/via/keymap.c:44]:
```suggestion
             KC_ESC,  KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6,           KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,    KC_BSPC,     KC_DEL,
            KC_PGUP,  KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T,                 KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS,
           KC_PGDN,   KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G,                KC_H, KC_J, KC_K, KC_L,  KC_SCLN,   KC_QUOT,   KC_ENT,
                      KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B,                KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP,
                      KC_LCTL,     KC_LALT, KC_SPC, KC_LGUI,                KC_SPC,     MO(1),                        KC_LEFT, KC_DOWN, KC_RGHT
    ),
    [1] = LAYOUT_full_bs(
             RESET,   _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,          KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______,     _______,
            RGB_MOD,  _______, KC_BRID, KC_BRIU, _______, _______, _______,                 _______, _______, _______, _______, _______, RGB_VAD, RGB_VAI, _______,
           RGB_TOG,   _______, KC_VOLD, KC_VOLU, KC_MUTE, _______, _______,                _______, _______, _______, _______,  _______,   _______,   _______,
                      _______, KC_MPRV, KC_MPLY, KC_MNXT, _______, _______,                _______, _______, _______, _______, _______, _______, _______, _______,
                      _______,     _______, _______, _______,                _______,     _______,                        _______, _______,_______ 
    ),
    [2] = LAYOUT_full_bs(
             _______,   _______, _______,   _______,   _______,   _______,   _______,   _______,          _______,   _______,   _______,   _______,  _______,  _______,  _______,     _______,
            _______,  _______, _______, _______, _______, _______, _______,                 _______, _______, _______, _______, _______, _______, _______, _______,
           _______,   _______, _______, _______, _______, _______, _______,                _______, _______, _______, _______,  _______,   _______,   _______,
                      _______, _______, _______, _______, _______, _______,                _______, _______, _______, _______, _______, _______, _______, _______,
                      _______,     _______, _______, _______,                _______,     _______,                        _______, _______,_______ 
    ),
    [3] = LAYOUT_full_bs(
             _______,   _______, _______,   _______,   _______,   _______,   _______,   _______,          _______,   _______,   _______,   _______,  _______,  _______,  _______,     _______,
            _______,  _______, _______, _______, _______, _______, _______,                 _______, _______, _______, _______, _______, _______, _______, _______,
           _______,   _______, _______, _______, _______, _______, _______,                _______, _______, _______, _______,  _______,   _______,   _______,
                      _______, _______, _______, _______, _______, _______,                _______, _______, _______, _______, _______, _______, _______, _______,
```

File: keyboards/bear_65/rules.mk [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
```
Comment (fauxpark) [inline keyboards/bear_65/rules.mk:21]:
```suggestion
```

File: keyboards/bear_65/info.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (fauxpark) [inline keyboards/bear_65/info.json:1]: Please regenerate this file without the rotation - it currently looks like this in Configurator:
image
You can test this out by pressing Ctrl+Shift+I to load a local info.json.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by MudkipMao** [comment]
@drashna why did you have me remove the key_count: field in the info.json? Here https://beta.docs.qmk.fm/developing-qmk/qmk-reference/reference_info_json it says that the field is required

**Comment by fauxpark** [comment]
IMO the docs are wrong on that point. `key_count` isn't really needed as you can determine the number of keys by just counting the number of elements in the `layout` array.

**Comment by noroadsleft** [comment]
> Hi QMK maintainers, one thing I do want to ask is how we want to structure this folder. Jacky is the designer of the S7_Elephant which currently has its own folder. Should I consolidate both of these boards under 1 folder?

It would be more straightforward to move both boards in a PR where neither of the codebases is changing. I vote to keep the boards in their own folders for now.

File: keyboards/bear_65/bear_65.h [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   KC_NO }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,   K1E   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   KC_NO, K2D,   K2E   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   K3E   }, \
+    { K40,   KC_NO, KC_NO, K43,   KC_NO, K45,   K46,   KC_NO, K48,   KC_NO, K4A,   K4B,   K4C,   K4D,   K4E   }, \
+}
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (noroadsleft) [inline keyboards/bear_65/bear_65.h:49]:
```suggestion
```

File: keyboards/bear_65/config.h [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+//     #define RGBLIGHT_EFFECT_SNAKE
+//     #define RGBLIGHT_EFFECT_KNIGHT
+//     #define RGBLIGHT_EFFECT_CHRISTMAS
+//     #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//     #define RGBLIGHT_EFFECT_RGB_TEST
+//     #define RGBLIGHT_EFFECT_ALTERNATING
+#endif
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (noroadsleft) [inline keyboards/bear_65/config.h:79]:
```suggestion
```

File: keyboards/bear_65/config.h [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+#define MATRIX_COL_PINS { D0, D1, D2, D3, D5, D4, D6, D7, B4, F7, F6, F5, F4, F1, F0 }
+
+/* Setting the matrix value of top left key for bootmagic lite */
+#define BOOTMAGIC_LITE_ROW      3
+#define BOOTMAGIC_LITE_COLUMN   1
+
+#define DIODE_DIRECTION ROW2COL
+
+#define BACKLIGHT_PIN B6
+// #define BACKLIGHT_BREATHING
```
Comment (noroadsleft) [inline keyboards/bear_65/config.h:53]: Backlight Breathing through Hardware PWM should be supported on this PIN, IIRC.

```suggestion
#define BACKLIGHT_BREATHING
```

File: keyboards/bear_65/info.json [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+    "keyboard_name": "Bear 65",
+    "url": "https://qmk.fm/keyboards",
+    "maintainer": "qmk",
+    "width": 19.75,
+    "height": 6.014620257184662,
```
Comment (noroadsleft) [inline keyboards/bear_65/info.json:6]:
```suggestion
    "width": 20.75,
    "height": 5,
```

File: keyboards/bear_65/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,39 @@
+/* Copyright 2020 REPLACE_WITH_YOUR_NAME
```
Comment (noroadsleft) [inline keyboards/bear_65/keymaps/default/keymap.c:1]:
```suggestion
/* Copyright 2020 MudkipMao
```

File: keyboards/bear_65/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,47 @@
+/* Copyright 2020 REPLACE_WITH_YOUR_NAME
```
Comment (noroadsleft) [inline keyboards/bear_65/keymaps/via/keymap.c:1]:
```suggestion
/* Copyright 2020 MudkipMao
```

File: keyboards/bear_65/readme.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
+# Bear 65
+
+![Bear 65](https://i.imgur.com/W1sVP1T.jpg)
```
Comment (noroadsleft) [inline keyboards/bear_65/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![Bear 65](https://i.imgur.com/W1sVP1Tl.jpg)
```

File: keyboards/bear_65/rules.mk [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+LTO_ENABLE = yes            # Use LTO flags to reduce firmware size
+
+# generated by KBFirmware JSON to QMK Parser
+# https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (noroadsleft) [inline keyboards/bear_65/rules.mk:27]:
```suggestion
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10384/files).

**Review by noroadsleft — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**